src/batched/batchedmul.jl

```diff
@@ -32,13 +35,13 @@ _unbatch(A::BatchedAdjOrTrans) = A.parent
 const _GemmFloat = Union{Float64, Float32, ComplexF64, ComplexF32}
 
 _BATCHED_GEMM_LIST = [
-    (:(StridedArray{T, 3}), 'N'),
-    (:(BatchedTranspose{T, <:StridedArray{T, 3}}), 'T'),
-    (:(BatchedAdjoint{T, <:StridedArray{T, 3}}), 'C')
+    (:(Array{T, 3}), 'N'),
+    (:(BatchedTranspose{T, <:Array{T, 3}}), 'T'),
+    (:(BatchedAdjoint{T, <:Array{T, 3}}), 'C')
```

Comment:
why do we have to restrict these? Were they causing ambiguities with the cuarrays ones from https://github.com/JuliaGPU/CuArrays.jl/pull/619? the latter seem stricter than the stridedarray versions here....

Reply:
Maybe there's no need. Does `StridedArray` sound right to you? 

In the opposite direction, I was wondering whether anything more general should be accepted. Here is one where `batched_gemm!` is happy to try but gives the wrong answer, I think:
```julia
julia> aa = rand(2,2,2); bb = rand(2,2,2);

julia> cc = PermutedDimsArray(bb, (1,3,2));

julia> strides(cc) # starts with 1
(1, 4, 2)

julia> dd = NNlib.batched_gemm!('N', 'N', 1.0, aa, cc, 0.0, similar(aa)) # runs
2×2×2 Array{Float64,3}:
[:, :, 1] =
 0.124876  0.271354
 0.324212  0.226239

[:, :, 2] =
 0.458338  2.53924e-315
 0.270784  1.06338e-314

julia> dd ≈ batched_mul(aa, cc) # fallback disagrees
┌ Debug: calling fallback method for batched_mul!
│   typeof(A) = Array{Float64,3}
│   typeof(B) = PermutedDimsArray{Float64,3,(1, 3, 2),(1, 3, 2),Array{Float64,3}}
│   typeof(C) = Array{Float64,3}
└ @ NNlib ~/.julia/dev/NNlib/src/batched/batchedmul.jl:66
false
```

Reply:
I've now fixed the above error with strides of `batched_gemm!`.